### ISSUE TITLE
Liquibase UUID type parsing as type="${uuidType}"> in Azure MariaDB instance

### ISSUE THREAD

**Issue by jsm174** [jhipster/generator-jhipster#9917]
##### **Overview of the issue**

We've been porting a 5.8.2 app to 6.1.0 and decided to use the awesome new `UUID` type (https://github.com/jhipster/generator-jhipster/pull/9794) 

Running locally in docker everything was working great:

```
      - 'SPRING_DATASOURCE_URL=jdbc:mariadb://blog-mariadb:3306/blog'
```

We finally started deploying to our Azure environment and thats when we had problems:

```
      - 'SPRING_DATASOURCE_URL=jdbc:mariadb://########.database.azure.com:3306/blog?useSSL=true&trustServerCertificate=true'
```

After turning on liquibase logging:

```
logging:
   level:
      ROOT: DEBUG
      liquibase: TRACE
```

we noticed the the `uuidType` was not being replaced with `varchar(36)`:

docker:

```
blog-app_1       |         [
blog-app_1       |             [
blog-app_1       |                 nullable="false"
blog-app_1       |             ]
blog-app_1       |             name="interest_uuid"
blog-app_1       |             type="varchar(36)"
blog-app_1       |         ],
blog-app_1       |         [
```

azure:

```
blog-app_1       |         [
blog-app_1       |             [
blog-app_1       |                 nullable="false"
blog-app_1       |             ]
blog-app_1       |             name="interest_uuid"
blog-app_1       |             type="${uuidType}"
blog-app_1       |         ],
```

Because of this issue, the app would fail to start:

```
blog-app_1       | 2019-06-15 03:13:34.843  WARN 1 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is java.lang.RuntimeException: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'servletEndpointRegistrar' defined in class path resource [org/springframework/boot/actuate/autoconfigure/endpoint/web/ServletEndpointManagementContextConfiguration$WebMvcServletEndpointManagementContextConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.endpoint.web.ServletEndpointRegistrar]: Factory method 'servletEndpointRegistrar' threw exception; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'auditEventsEndpoint' defined in class path resource [org/springframework/boot/actuate/autoconfigure/audit/AuditEventsEndpointAutoConfiguration.class]: Unsatisfied dependency expressed through method 'auditEventsEndpoint' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'customAuditEventRepository' defined in file [/app/classes/com/example/blog/repository/CustomAuditEventRepository.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'persistenceAuditEventRepository': Cannot create inner bean '(inner bean)#d5f3118' of type [org.springframework.orm.jpa.SharedEntityManagerCreator] while setting bean property 'entityManager'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name '(inner bean)#d5f3118': Cannot resolve reference to bean 'entityManagerFactory' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [com/example/blog/config/LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
```
 
A colleague decided to update `master.xml` from:

```
<property name="uuidType" value="varchar(36)" dbms="h2, mariadb"/>
```

to

```
<property name="uuidType" value="varchar(36)" dbms="mysql, h2, mariadb"/>
```

Sure enough that worked! 

So for some reason, Azure mariadb is reporting itself as mysql.

I'm not sure if `mysql` should ultimately be added to the generator, or if this is some obscure Azure issue, but I thought it should be reported, incase anyone else has the same issue.

##### **Motivation for or Use Case**

Can't get it running in azure without the fix.

##### **Reproduce the error**

This is the JDL use to reproduce error:

```
application {
   config {
      baseName gateway,
      applicationType gateway,
      authenticationType oauth2,
      packageName com.example.gateway,
      databaseType no
      buildTool gradle
   }
   entities *
}

application {
   config {
      baseName blog,
      applicationType microservice,
      authenticationType oauth2,
      packageName com.example.blog,
      prodDatabaseType mariadb,
      buildTool gradle
   }
   entities BlogUserInterest
}

entity BlogUserInterest {
   interestUuid UUID required,
   createdAt Instant,
   lastUpdate Instant
}

paginate * with pagination

microservice BlogUserInterest with blog
``` 

##### **Suggest a Fix**

When UUIDs are used, add `mysql` to liquibase `master.xml`:

```
<property name="uuidType" value="varchar(36)" dbms="mysql, h2, mariadb"/>
```

##### **JHipster Version(s)**

```
blog@0.0.0 /Users/jmillard/git/new-jh/blog
└── generator-jhipster@6.1.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": [
      "en",
      "fr"
    ],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "com.example.blog",
    "packageFolder": "com/example/blog",
    "prodDatabaseType": "mariadb",
    "searchEngine": false,
    "serviceDiscoveryType": "eureka",
    "skipClient": true,
    "testFrameworks": [],
    "websocket": false,
    "baseName": "blog",
    "applicationType": "microservice",
    "authenticationType": "oauth2",
    "buildTool": "gradle",
    "jhipsterVersion": "6.1.0",
    "skipUserManagement": true,
    "clientPackageManager": "npm",
    "cacheProvider": "hazelcast",
    "serverPort": "8081",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": []
  },
  "entities": [
    "BlogUserInterest"
  ]
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity BlogUserInterest {
  interestUuid UUID required,
  createdAt Instant,
  lastUpdate Instant
}
paginate BlogUserInterest with pagination
microservice BlogUserInterest with blog

</pre>
</details>

##### **Environment and Tools**

openjdk version "12.0.1" 2019-04-16
OpenJDK Runtime Environment AdoptOpenJDK (build 12.0.1+12)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 12.0.1+12, mixed mode, sharing)

git version 2.20.1 (Apple Git-117)

node: v10.15.3

npm: 6.4.1

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

INFO! Congratulations, JHipster execution is complete!

##### **Browsers and Operating System**

-   [X] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
Thanks for reporting this but you still need to provide us with all details we ask in issue template without those we may not be able to reproduce this and hence the issue will be closed. So please update the issue with your .yo-rc.json and entity JDL. Run jhipster info and paste output here

**Comment by jsm174** [comment]
Sorry @deepu105, issue updated accordingly.

**Comment by deepu105** [comment]
Thank you

**Comment by murdos** [comment]
@jsm174 : thanks for detailed report, and the analysis of the issue! I'm also glad you enjoy the new UUID support. 
Do you want to fix the issue yourself with a PR? 
Changing this line should be enough: https://github.com/jhipster/generator-jhipster/blob/master/generators/server/templates/src/main/resources/config/liquibase/master.xml.ejs#L30 - I've used `prodDatabaseType` because it sounded smarter, but it seems we should explicitly use `dbms="h2, mysql, mariadb"` when mysql or mariadb is used.

**Comment by jsm174** [comment]
Sure. I can fix it!

**Comment by jsm174** [comment]
@murdos - Create PR https://github.com/jhipster/generator-jhipster/pull/9927

Thanks!

**Comment by pascalgrimaud** [comment]
Fixed by https://github.com/jhipster/generator-jhipster/pull/9927

**Comment by romarioliveira25** [comment]
> Fixed by #9927

Same issue on oracle database? Needs the same fix?
